### ISSUE TITLE
Samba share changes in CE

### ISSUE THREAD

**Issue by Nostalgist** [CoreELEC/CoreELEC#14]
Hi there!
Today I updated from LE-8.90.6 to CE-8.90.0 (made a LE backup right before that)
My sources are defined in the sources.xml file with username and password included:
smb://user:pass@IP/source/
After the update I tried to access one of those sources but CE asked me for user and pass. I entered them and clicked on "save". Unfortunately the watched status of all the files of the source are now gone.
I tried restoring (still in CE), but that just made it happen again.
I guess I can just downgrade and restore. (will try that the next days)

Any idea what might have changed between LE-8.90.6 and CE-8.90.0 that could cause this? I saw in the commits that the samba version was updated recently.

I don't think this is much helpful, but anyway: https://pastebin.com/RFkkixiF

**Comment by adamg88** [comment]
Your logs just indicate that the username/password is incorrect and as such Kodi is not scanning for files in the directory, I would try to use NFS as there is a big performance hit when using Samba on low powered devices.

**Comment by Nostalgist** [comment]
Thank your for looking at my logs and the advice.
I actually don't have any issues with Samba performance.

However, I managed to determine and solve the problem. It seems like the format "smb://user:pass@IP/source/" is no longer properly supported (I did this since the days of XBMC on the XBOX classic :-) ). Instead the user and pass needs to be set in passwords.xml (this must have been changed between LE-8.90.6 and CE-8.90.6). This causes a lot of trouble and srews up the whole database. Anyway, if someone is experiencing the same problem, here is what I did (replace user, pass and IP with your own):
1. Downgrade to LE-8.90.6, restore Backup --> everthing working again
2. Create/Edit passwords.xml:
`<passwords>
    <path>
        <from pathversion="1">smb://IP/</from>
        <to pathversion="1">smb://user:pass@IP/</to>
    </path>
</passwords>`
3. Edit sources.xml: remove "user:pass@" in every entry
4. Open MyVideos109.db in DB Browser for SQL Lite https://github.com/sqlitebrowser/sqlitebrowser/releases
and run:
`update path
set strPath = replace(strPath, 'user:pass@', '');
update movie
set c22= replace(c22, 'user:pass@', '');
update files
set strfilename= replace(strfilename, 'user:pass@', '');
update episode
set c18= replace(c18, 'user:pass@', '');`
5. Move files to box
6. Delete Thumbnails folder and Textures13.db, then reboot
6. Watched status of files still OK? --> start update to Coreelec
7. Be happy
